Сделал опциональный проект Outsmart в desktop версии

kirill-io opened the pull request:
Сделал опциональный проект в desktop версии, прощу проверить. 
Некоторые svg-картинки подключены инлайном из внешнего файла и поэтому отображаются на сервере или при запуске сервера, если просто открыть index.html они не отобразятся.

kirill-io committed:
feat: Добавлена папка image, заполнен head

Добавил в корень проекта папку image, где будут хранится изображения использованные в проекте.
Дабавлена и подключена favicon.
Заполнен блок head в index.html

kirill-io committed:
feat: Добавлен normalize

В проект добавлен и подключен normalize.css

kirill-io committed:
feat: Добавлены шрифты

Добавлены и подключены шрифты к проету

kirill-io committed:
feat: добавлена секция promo в desktop версии

Сверстана секция promo в desktop версии
Добавлены стили по состоянию у элементов <a> и <button>

kirill-io committed:
feat: добавлена секция coverage в desktop версии

Сверстана секция coverage в desktop версии

kirill-io committed:
feat: добавлена секция services в desktop версии

Сверстана секция services в desktop версии
Добавлены состояния по hover

kirill-io committed:
feat: добавлена секция approach в desktop версии

Сверстана секция approach в desktop версии

kirill-io committed:
feat: добавлена секция certifications в desktop версии

Сверстана секция certifications в desktop версии

kirill-io committed:
feat: добавлена секция afterword в desktop версии

Сверстана секция afterword в desktop версии

kirill-io committed:
Добавлена секция outsmart в desktop версии

Сверстана секция outsmart в desktop версии

kirill-io committed:
feat: Добавлены секции Portfolio, Customers, Touch, Footer в desktop версии

Сверстаны секции Portfolio, Customers, Touch, Footer в desktop версии

c2u5hed commented:
1. Проект не запускается командой yarn dev из-за ошибки в маршрутах css-файлов
2. Оверлеи для первых двух секций необходимо реализовать векторными, а не растровыми изображениями, а также не через background-image, а через отдельный абсолютно-спозиционированный слой
3. В секции Product developers нет отступа
4. В форме у состояния Focused неверная стилизация. При появлении границы при попадании элемента в фокус текст внутри поля смещается
5. На домашней странице сайта логотип следует реализовывать без ссылки — правило хорошего тона по SEO (страница не должна вести сама на себя)
6. Для всех растровых изображений на сайте не реализована х2 версия для Retina-дисплеев
7. Для всех растровых изображений на сайте не реализован формат .webp
8. Логотипы сертификаций и клиентов следует реализовать векторной графикой
9. Расположение логотипа WorkflowEngine в карточках Portfolio не соответствует дизайн-макету
10. Отступы в карточках реализованы через margin: 0 auto и max-width, а должны быть через padding и нужные размеры внутреннего контента
11. У копирайта отсутствует год

janguardian commented on index.html:
```diff
@@ -1,12 +1,499 @@
+      </div>
+    </div>
+
+    <aside class="coverage wrapper" id="coverage">
```
Почему `<aside>` а не `<section>`?

kirill-io replied:
Подумал, что так будет семантичнее, переделаю

janguardian commented on index.html:
```diff
@@ -1,12 +1,499 @@
+      </div>
+    </div>
+
+    <aside class="coverage wrapper" id="coverage">
```
Все секции и их контейнеры реализованы одинаково с небольшими вариациями. Почему бы не сделать один блок с модификаторами?

kirill-io replied:
Понял, переделаю

janguardian commented on index.html:
```diff
@@ -1,12 +1,499 @@
-    <p class="paragraph">Hello, World!</p>
+  <body class="root">
+    
+    <div class="promo wrapper">
```
Почему не `<section>`?

kirill-io replied:
Иначе валидатор ругается, что в потомках нет заголовка, изначально тут была секция.

janguardian commented on index.html:
```diff
@@ -1,12 +1,499 @@
+            </nav>
+          </div>
+        </header>
+        <section class="promo__content">
```
Почему здесь `<section>`?

kirill-io replied:
Иначе валидатор ругается, что в потомках нет заголовка, изначально тут был div

janguardian commented on index.html:
```diff
@@ -1,12 +1,499 @@
+      </div>
+    </section>
+
+    <article class="afterword wrapper" id="afterword">
```
Почему `<article>`?

kirill-io replied:
Подумал, что так будет семантичнее, переделаю

janguardian reviewed (changes requested):
1. Проект не запускается командой yarn dev из-за ошибки в маршрутах css-файлов
2. Оверлеи для первых двух секций необходимо реализовать векторными, а не растровыми изображениями, а также не через background-image, а через отдельный абсолютно-спозиционированный слой
3. В секции Product developers нет отступа
4. В форме у состояния Focused неверная стилизация. При появлении границы при попадании элемента в фокус текст внутри поля смещается
5. На домашней странице сайта логотип следует реализовывать без ссылки — правило хорошего тона по SEO (страница не должна вести сама на себя)
6. Для всех растровых изображений на сайте не реализована х2 версия для Retina-дисплеев
7. Для всех растровых изображений на сайте не реализован формат .webp
8. Логотипы сертификаций и клиентов следует реализовать векторной графикой
9. Расположение логотипа WorkflowEngine в карточках Portfolio не соответствует дизайн-макету
10. Отступы в карточках реализованы через margin: 0 auto и max-width, а должны быть через padding и нужные размеры внутреннего контента
11. У копирайта отсутствует год
12. В проекте следует обойтись без файла нормализации, особенно учитывая что он содержит очень много неприменимого к проекту кода.

kirill-io committed:
fix: Исправлена ошибка yarn

Исправлена ошибка связанная с неправильной маршрутизацией css-файлов и запущен yarn

kirill-io committed:
refactor: Произведен рефакторонг кода в index.html

Тег div секции promo заменен на тег section. А также в секции promo поменял местами блоки promo__content и header, чтобы исправить предупреждение валидатора.
Тег aside секции coverage изменен на тег section.
Тег article секции afterword изменен на тег section

kirill-io committed:
style: Изменены стили блока promo__container

После рефакторинга кода в секции promo изменены стили блока promo__container для соответствия с макетом

kirill-io committed:
refactor: Произведен рефакторонг кода в index.html

Реорганизованы классы у тегов section в блоки wrapper с соответствующими модификаторами.
Реорганизованы блоки container, добавлены соответствующие модификаторы.
Модифицированы файлы в директории blocks в соответствии с реорганизацией классов

kirill-io committed:
fix: Исправлена секция promo.

Overlay секции promo исправлен, подключен инлайноми и используется векторное изображение.
Логотип сайта реализован без ссылки

kirill-io committed:
fix: Исправлена секция coverage.

Overlay секции coverage исправлен, подключен инлайноми и используется векторное изображение

kirill-io committed:
refactor: Поправлен отступ в карточке Product developers секции coverage

Добавлен отсутствующий div с классом coverage__content в карточку Product developers

kirill-io committed:
style: Поправлена стилизация input в форме при focus.
>>
>> Свойствo border при focus на input заменено на outline, добавлено свойство box-shadow.
Заменил png-иконку close-icon на svg-иконку

kirill-io committed:
fix: Произведены изменения в секции promo

Блок logo подключен к странице через тег img.
Overlay добавлен в формате svg и подключен через тег img.
Переделаны кнопки в секции promo, а именно обернуты в div.
Подготовлены и подключены изображения в формате webp.
Подготовлены и подключены изображения для Retina-дисплеев

kirill-io committed:
style: В секцию coverage добавлены изображения webp и x2 версия

В секцию coverage добавлены и подключены фоновые изображения в формате webp и x2 версия для Retina дисплеев

kirill-io committed:
style: В секцию services добавлены изображения webp и x2 версия

В секцию services добавлены и подключены фоновые изображения в формате webp и x2 версия для Retina дисплеев

kirill-io committed:
fix: Переделана секция approach

В изображениях фоновое изображение добавлено фоном, svg изображение добавлено на страницу через тег img. Также добавлен формат webp для фоновых изображений и реализована x2 версия для Retina-дисплеев.
Блок numbered-cards переделан с grid позиционирования на flex позиционирование

kirill-io committed:
fix: Убран горизонтальный скролл в секциях promo и coverage.

Убран появляющийся горизонтальный скролл в секциях promo и coverage

kirill-io committed:
fix: Переделана секция coverage

Блок coverage переделан с grid позиционирования на flex позиционирование

kirill-io committed:
fix: Изменены логотипы секции certifications

Логотипы секции certifications добавлены в виде svg-файлов

kirill-io committed:
fix: Изменено изображение секции afterword

Для изображения секции afterword подключен формат webp и реализован вариант для Retina-дисплеев

kirill-io committed:
fix: Изменена секция outsmart

В секции outsmart переделано позиционирование с grid на flex.
Изобрабражения добавлены в формате svg

kirill-io committed:
fix: Изменена секция portfolio

Изображения в секции добавлены в формате svg.
Изменена позиция логотипов, логотипы подняты на 5px

kirill-io committed:
fix: Изменена секция customers.

Изображения в секции добавлены в формате svg.
Позиционирование элементов изменено с grid на flex

kirill-io committed:
fix: Изменена секция touch.

Фоновое изображение добавлено в формате webp и для retina дисплеев

kirill-io committed:
fix: Изменена секция footer.

В контент блока footer__copyright добавлен год.
В блок footer__row-bottom добавлен верхний отступ

kirill-io committed:
fix: Из проекта удален normalize

kirill-io committed:
feat: Сборка проекта
